### PR TITLE
[Doc] Add onboarding guide

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

and link onboarding guide from README, CONTRIBUTING

### Issues Resolved

fixes https://github.com/opensearch-project/.github/issues/131

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ONBOARDING.md
Comment (joshuarrrr) [inline ONBOARDING.md:38]: I would have preferred to just link to this as a reading, but it's pretty apache-specific and didn't capture the ways we've adapted for our project. But I could imagine this info living elsewhere. It's worth noting that this section is not original work, but I believe I've adapted and referenced it appropriately.

File: ONBOARDING.md
Comment (joshuarrrr) [inline ONBOARDING.md:70]: Ideally we could just point to a single GitHub doc. But the GitHub getting started doc has a lot of extra content and noise that distract from the task. So I decided to go with a stripped down list here.

File: ONBOARDING.md
Comment (dblock) [inline ONBOARDING.md:13]: Nit: can we Capitalize Titles Like in Other Markdown Files?

**Review by dblock — COMMENTED** [review]
I like this very much. 

@CEHENKLE Did we have another one of these somewhere? 

@joshuarrrr Do we link this from https://github.com/opensearch-project/.github/blob/main/profile/README.md?

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:13]: lol, sure - I'll also double check the style guide for other things I may have missed. Sentence case titles are a strong personal preference, but consistency is better 😉

Comment (AMoo-Miki) [reply ONBOARDING.md:13]: At least the CONTRIBUTING uses cap-titles.

File: ONBOARDING.md
Comment (AMoo-Miki) [inline ONBOARDING.md:34]: I know it is the title of that article but as an adjective for "Elasticsearch", the term should be hyphenated as "open-source"; I don't think we should change it here unless they fix the origin.

File: ONBOARDING.md
Comment (AMoo-Miki) [inline ONBOARDING.md:56]: Should this be more than 100?

File: ONBOARDING.md
Comment (AMoo-Miki) [reply ONBOARDING.md:70]: maybe put a link to that one to say if you need more information, see a complete version there.

File: ONBOARDING.md
Comment (AMoo-Miki) [inline ONBOARDING.md:75]: If the email is not made public, they cannot sign DCO. Evidenced by the number people who have missed this, IMO, we should point it out explicitly.

File: ONBOARDING.md
Comment (AMoo-Miki) [inline ONBOARDING.md:82]: They are called "forum" on top and "forums" here; which one is it?

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:13]: I'll defer to maintainers of this repo over whether we choose title casing to match other markdown files here, or whether it's better for new documents to follow the project style guide: https://github.com/opensearch-project/documentation-website/blob/main/STYLE_GUIDE.md#punctuation-and-capitalization

> Use sentence case for titles, headings, and table headers. Titles of standalone documents may use title case.

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:19]: Huh, interesting. Seems like a bug/quirk in Quip's markdown export. Will replace.

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:34]: yeah, agreed, on both points. My intention here is to match the actual document titles of the source.

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:56]: This is only counting public repos (it excludes archived repos), which is currently 97. But the link is wrong. I'll update to https://github.com/orgs/opensearch-project/repositories?q=&type=public&language=&sort=

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:74]: Sure, will add.

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:75]: Is it really a requirement for contribution? Any developer who would prefer not to disclose their email publicly is de facto barred from contributing? If so, I definitely agree we need to highlight that (not just here, but in the DCO section of CONTRIBUTING)

File: ONBOARDING.md
Comment (joshuarrrr) [reply ONBOARDING.md:82]: We're not consistent and use both interchangeably in this repo: https://github.com/search?q=repo%3Aopensearch-project%2F.github%20forum&type=code
Same thing in the style guide: https://github.com/opensearch-project/documentation-website/blob/main/STYLE_GUIDE.md#brand-personality-traits

If it's up to me, I prefer the singular and will update accordingly.

**Comment by joshuarrrr** [comment]
> @joshuarrrr Do we link this from https://github.com/opensearch-project/.github/blob/main/profile/README.md?

Added link in most recent commit.

**Review by peternied — DISMISSED** [review]
Thanks for working on this @joshuarrrr.  Looking at the unresolved threads most of these seems subtle, if there are strong opinions, lets follow up up with additional pull requests.

File: ONBOARDING.md
Comment (peternied) [reply ONBOARDING.md:13]: I'd be happy to review a PR for a markdown linter if we'd like to codify these styles

**Review by peternied — APPROVED** [review]

File: CONTRIBUTING.md
Comment (ananzh) [inline CONTRIBUTING.md:97]: should be `--signoff` not `––signoff` right?

File: ONBOARDING.md
Comment (ananzh) [inline ONBOARDING.md:43]: `a voluntary basis`?

File: ONBOARDING.md
Comment (mch2) [inline ONBOARDING.md:50]: I think we should link the [OpenSearch code of conduct](https://opensearch.org/codeofconduct.html).

**Comment by dblock** [comment]
@joshuarrrr finish this?

**Comment by AMoo-Miki** [comment]
How would you feel about merging this and I will follow it up with the 2 changes asked for?

**Comment by dblock** [comment]
Let's close this one - @AMoo-Miki reopen a new one on top?

**Pull request closed without merging**
